modules/liveintentAnalyticsAdapter.js

```diff
@@ -0,0 +1,157 @@
+import {ajax} from '../src/ajax.js';
+import { generateUUID, logInfo } from '../src/utils.js';
+import adapter from '../libraries/analyticsAdapter/AnalyticsAdapter.js';
+import CONSTANTS from '../src/constants.json';
+import adapterManager from '../src/adapterManager.js';
+import { auctionManager } from '../src/auctionManager.js';
+
+const analyticsType = 'endpoint';
+const url = 'https://wba.liadm.com/analytic-events';
+const gvlid = 148;
+const adapterCode = 'liveintent';
+const { EVENTS: { AUCTION_END } } = CONSTANTS;
+let initOptions = {}
+
+let liAnalytics = Object.assign(adapter({url, analyticsType}), {
+  track({ eventType, args }) {
+    if (typeof args !== 'undefined') {
+      switch (eventType) {
+        case AUCTION_END:
+          liAnalytics.handleAuctionEnd(args);
+          break;
+        default: break;
+      }
+    }
+  }
+});
+
+liAnalytics.handleAuctionEnd = function(args) {
+  const bidWonTimeout = (initOptions && initOptions.bidWonTimeout) || 2000;
+  const sampling = (initOptions && initOptions.sampling) || 0.1;
+  const isSampled = Math.random() < parseFloat(sampling)
```

Comment:
Let's not sample every individual event, but determine once on page load whether we are sampled. This should lead to more consistent data

Reply:
➕